controllers/admin-api-controller.js

```diff
@@ -0,0 +1,242 @@
+            description,
+            products,
+            listed,
+            featured
+        })
+        category = await category.save()
+        res.json({ 
+            success: true,
+            category: categoryTemplate(category)
+          })
+    }catch(error){
+        stdRes._500(res, error.message)
+    }
+}
+
+exports.patchCategory = async (req, res, next) => {
+    try{
+        const categoryId = req.params.categoryId
+    
+        const category = await Category.findByIdAndUpdate(categoryId, req.body, {new:true, upsert:true})
+        res.json({ 
+            success: true,
+            category: categoryTemplate(category)
+        })
+    }catch(error){
+        stdRes._500(res, error.message)
+    }
+}
+
+exports.deleteCategory = async (req, res, next) => {
+    try{
+        const categoryId = req.params.categoryId
+    
+        await Category.findByIdAndDelete(categoryId)
+        res.json({ success: true })
+    }catch(error){
+        stdRes._500(res, error.message)
+    }
+}
+
+// Product 
+exports.postProduct = async (req, res, next) => {
+    try{
+        const title = req.body.title
+        const description = req.body.description
+        const price = req.body.price
+        const compareAt = req.body.compareAt
+        const imagePaths = req.body.imagePaths
+        const categories = req.body.categories
+        const shippingCost = req.body.shippingCost
+        const stock = req.body.stock
+        const publish = req.body.publish
+      
+        let product = new Product({
+            title,
+            description,
+            price,
+            compareAt,
+            imagePaths,
+            categories,
+            shippingCost,
+            stock,
+            publish
+        })
+        product = await product.save()
+        product.categories(categories)
+        res.json({ 
+            success: true,
+            product: productTemplate(product)
+          })
+    }catch(error){
+        stdRes._500(res, error.message)
+    }
+}
+
+exports.patchProduct = async (req, res, next) => {
+    try{
+        const productId = req.params.productId
+    
+        const product = await Product.findByIdAndUpdate(productId, req.body, {new:true, upsert:true})
+        res.json({ 
+            success: true,
+            product: productTemplate(product)
+        })
+    }catch(error){
+        stdRes._500(res, error.message)
+    }
+}
+
+exports.deleteProduct = async (req, res, next) => {
+    try{
+        const productId = req.params.productId
+    
+        await Product.findByIdAndDelete(productId)
+        res.json({ success: true })
+    }catch(error){
+        stdRes._500(res, error.message)
+    }
+}
+
+// User 
+exports.postUser = async (req, res, next) => {
+    try{
+        const firstName = req.body.firstName
+        const lastName = req.body.lastName
+        const email = req.body.email
+        const admin = req.body.admin
+        const password = req.body.password
```

Comment:
You didnt include the owner parameter. Its necessary to check if the logged-in user is an owner.